### PR TITLE
feat(cli): add standalone agent-friendly CLI for `tools` 

### PR DESCRIPTION
**Pull request by zew1me**
## Summary

- ship `icuvisor-cli` alongside the MCP binary using the existing build, signing, packaging, and release flow
- expose the tools-only MVP through namespaced `tools list`, `tools describe`, and `tools call` commands, plus `doctor` and `capabilities`
- share local-athlete routing, registration gates, public-error sanitization, and per-call panic recovery between MCP and direct invocation
- add compact progressive catalog discovery, canonical MCP descriptors, stdin JSON arguments, and deterministic machine-readable process errors
- address review feedback for command parsing, local-athlete validation classification, Windows installer metadata, and public-core verification guidance

Resources and Prompts are explicitly reported as unsupported surfaces in this tools-only MVP and remain follow-up work.

## Process contract

- successful commands write JSON to stdout only
- failures leave stdout empty and write one JSON error to stderr
- exit code `2` identifies usage errors; exit code `1` identifies runtime errors
- API credentials remain configuration/keychain inputs and are never accepted as CLI flags or tool arguments
- effective coach mode fails closed because the standalone CLI is local-athlete-only

## Validation

- `make build`
- `go test ./...`
- `make test-race`
- `make lint`
- `make docs-tools` (clean catalog output)
- `git diff --check`

## Live CLI evidence

Using locally configured development intervals.icu credentials, these commands all returned exit `0`, valid JSON on stdout, and empty stderr:

```text
icuvisor-cli doctor
icuvisor-cli tools call get_athlete_profile --args '{}'
icuvisor-cli tools call get_today --args '{}'
```

The terse `get_today` response contained the expected sections and metadata:

```json
{
  "date": "2026-07-21",
  "fitness": [],
  "completed_activities": [],
  "planned_events": [],
  "wellness": [{
    "ctl": 41.386703,
    "atl": 28.980162,
    "hrv": 40,
    "restingHR": 49,
    "sleepQuality": 2,
    "sleepScore": 87,
    "sleepSecs": 29160,
    "steps": 8979
  }],
  "weather": {"status": "forecast_unavailable"},
  "_meta": {"timezone": "US/Pacific", "units": {"system": "metric"}}
}
```

### CODE REVIEW

File: pkg/icuvisor/icuvisor.go [hunk 1]
```diff
@@ -199,6 +203,122 @@ type Server struct {
...
+// Invoke calls one registered tool without an MCP transport.
+func (i *Invoker) Invoke(ctx context.Context, name string, arguments json.RawMessage) (ToolResult, error) {
+	if i == nil {
+		return ToolResult{}, errors.New("invoking tool: nil invoker")
+	}
+	tool, ok := i.tools[name]
+	if !ok {
+		return ToolResult{}, fmt.Errorf("unknown or unavailable tool %q", name)
+	}
+	outcome := toolexec.Execute(ctx, tool, tools.Request{Name: name, Arguments: arguments}, i.athleteID)
```
Comment (ricardocabral) [inline pkg/icuvisor/icuvisor.go:297]: **Keep direct-call errors diagnosable**

This invokes the shared executor, but direct invocations have no logger or `RecentToolCallRecorder`; MCP calls do. A CLI user can therefore receive the generic “check icuvisor logs” error after a handler failure or panic while the underlying cause was never logged.

That breaks the execution-boundary contract we agreed in #45 (logging/redaction belongs with the shared path) and leaves direct CLI failures materially harder to support. Please move the redacted failure logging/recording into the shared execution boundary, or inject the equivalent dependencies into `Invoker`, while preserving the stdout/stderr process contract.

File: pkg/icuvisor/icuvisor.go [hunk 2]
```diff
@@ -199,6 +203,122 @@ type Server struct {
...
+// InvokerOptions configures direct core-tool invocation for local CLI and agent views.
+type InvokerOptions struct {
+	Config     Config
+	Registry   Registry
+	Version    string
+	DeleteMode DeleteMode
+	Toolset    Toolset
+}
+
+// NewInvoker constructs a direct invoker from the same registry and safety policy used by MCP.
```
Comment (ricardocabral) [inline pkg/icuvisor/icuvisor.go:225]: **Suggestion (follow-up): lock the direct view to the MCP catalog with a parity matrix**

Now that `NewInvoker` is a second registration/invocation adapter, add a table-driven parity test across `compact`, `core`, and `full` toolsets and `none`, `safe`, and `full` delete modes. For each combination, compare the active tool names and descriptors (`name`, description, schemas, annotations, and catalog hash) with the MCP catalog; also cover local-athlete routing plus representative success, public-error, and panic calls, allowing only the intentional CLI omission of the MCP envelope.

A useful matrix would be:

| Policy / behavior | Assert |
| --- | --- |
| `compact` / `core` / `full` toolsets | The same visible tool names and compact metadata |
| `none` / `safe` / `full` delete modes | The same read/write/delete availability |
| Local-athlete mode | The same `athlete_id` rejection and configured-target routing |
| Every exposed tool | CLI `describe` matches MCP `name`, description, schemas, annotations, and catalog hash |
| Representative success/error/panic calls | The same shaped result or sanitized public error, except for CLI’s intentional omission of the MCP envelope |

This is not a merge blocker, but it is worth adding soon. It turns the #45 design promise—one safety-gated core contract with two views—into a regression guard, so future tool, gate, or schema changes cannot update MCP while leaving the CLI stale.
